app/helpers/codelist_helper.rb

```diff
@@ -170,4 +170,20 @@ def organisation_type_options
   def language_code_options
     Codelist.new(type: "language_code").to_objects
   end
+
+  def beis_allowed_channel_of_delivery_codes
+    Codelist.new(type: "channel_of_delivery_code", source: "beis").list
+  end
+
+  def channel_of_delivery_codes
+    iati_data = Codelist.new(type: "channel_of_delivery_code")
+    iati_data.select { |item| item["code"].in?(beis_allowed_channel_of_delivery_codes) }
```

Comment:
This is calling `beis_allowed_channel_of_delivery_code` once for every entry in the `channel_of_delivery_code` codelist. Because it isn't memoised, it's creating a new array 414 times (one of each entry in the IATI list), which means any test which hits the Activity details page is taking 30 seconds to run!

Reply:
Does `Codelist` memoise the read from disk during tests? Is it reading from disk that's expensive, or allocating the array in memory each time?